File: core/src/main/java/re/neotamia/config/annotation/ConfigProperty.java
package re.neotamia.config.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.FIELD)
public @interface ConfigProperty {
    /**
     * The description of the property.
     * Output as a comment above the property in the file if the format supports it.
     * @return the description
     */
    String value() default "";

    String name() default "";

    boolean exclude() default false;
}

File: core/src/main/kotlin/re/neotamia/config/annotation/ConfigProperty.kt
package re.neotamia.config.annotation

@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY)
/**
 * @param description A description of the configuration option. Output as a comment in the file if the format supports it.
 * @param name The name of the configuration option. If empty, the field or property name is used.
 * @param exclude If true, the field or property is excluded from serialization and deserialization.
 */
annotation class ConfigProperty(val description: String = "", val name: String = "", val exclude: Boolean = false)
